### ISSUE TITLE
Long delay during Generation of Personal Acess Tokens

### ISSUE THREAD

**Issue by omarmarquez** [gocd/gocd#6228]
When clicking on the [Generate] button nothing happens, the page stays the same, the javascript console shows no activity associated with the click action...

**Comment by omarmarquez** [comment]
Running haveged on startup did not alleviated the situation ... it might also been that haveged needs to be executed with different parameters (-F ? ) and then the container might need to run in privileged mode?

What seems to work fine was in the node (host) installing the  rng-tools package and the run
`$ sudo rngd -r /dev/urandom`
followed by a restart of the gocd container. 
With this the token generation response is generated immediately:

```
$ date && time curl -k https://chd-dev-vlog01:8154/go/api/current_user/access_tokens       -u admin:secret        -H 'Accept: application/vnd.go.cd.v1+json'        -X POST       -H 'Content-Type: application/json'       -d '{
        "description": "timing token creation with docker server "
      }' && date
Mon Apr 29 13:33:34 CDT 2019
{
  "_links" : {
    "self" : {
      "href" : "https://chd-dev-vlog01:8154/go/api/current_user/access_tokens/260"
    },
    "doc" : {
      "href" : "https://api.gocd.org/19.2.0/#access-tokens"
    },
    "find" : {
      "href" : "https://chd-dev-vlog01:8154/go/api/current_user/access_tokens/:id"
    }
  },
  "id" : 260,
  "description" : "timing token creation with docker server ",
  "username" : "admin",
  "revoked" : false,
  "revoke_cause" : null,
  "revoked_by" : null,
  "revoked_at" : null,
  "created_at" : "2019-04-29T18:33:34Z",
  "last_used_at" : null,
  "token" : "607cffc56afc3c513f15af1415141ae0a61e8d07"
}
real	0m0.143s
user	0m0.041s
sys	0m0.066s
Mon Apr 29 13:33:34 CDT 2019

```

Since this effectively resolves my problem I think is Ok to close this issue. 
Switching to use a non-blocking generator could also be a good thing to prevent it from happening in the future.

**Comment by ketan** [comment]
I've kicked off a build for my PR. This is a commit on top of version 19.2, with just one commit on it. If you feel comfortable using that docker image, I can point you to a URL that

> What seems to work fine was in the node (host) installing the rng-tools package and the run
> `$ sudo rngd -r /dev/urandom`

Thank you for letting us know. I've merged #6229 to not have this issue again. This change will move random number generator to use `/dev/urandom` instead of `/dev/random`. It seems acceptable (to me) to use a pseudo random number generator instead of a true random number generator.

Once you move over to version 19.4,  you can remove the rng-tools package and kill the rng daemon.
